### PR TITLE
Police log

### PR DESCRIPTION
**Pull request by ksavchyn**
Code to filter police incident data and create filtered table in database. Filters include:
-Nightlife
-Location
-Crime-Type

### CODE REVIEW

File: src/incidents_filtered.Rmd [hunk 1]
```diff
@@ -0,0 +1,87 @@
+---
+title: "incidents_filtered"
+author: "Kateryna Savchyn"
+date: "6/11/2019"
```
Comment (Nosferican) [inline src/incidents_filtered.Rmd]: Date should be in YYYY-MM-DD ([ISO_8601](https://en.wikipedia.org/wiki/ISO_8601)).

File: src/incidents_filtered.Rmd [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  library(pkg, character.only = TRUE)
+}
+```
+
+## Helper Functions
+```{r read-in incidents}
+get_incidents = function() {
+  conn <- dbConnect(drv = PostgreSQL(),
+                    dbname = "acpd",
+                    host = "postgis",
```
Comment (Nosferican) [inline src/incidents_filtered.Rmd]: That should be `host = postgis_1`

File: src/incidents_filtered.Rmd [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+```
+
+## Function to Pull, Filter, and Write Incidents data
+```{r write out filtered incidents}
+
+refresh_filtered_incidents <- function(){
+  # add stop block that throws error if table doesn't exist
+  conn <- dbConnect(drv = PostgreSQL(),
+                    dbname = "acpd",
+                    host = Sys.getenv("host"),
```
Comment (Nosferican) [inline src/incidents_filtered.Rmd]: Let's keep it with host `postgis_1`

File: src/incidents_filtered.Rmd [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                    port = Sys.getenv("port"),
+                    user = Sys.getenv("db_userid"),
+                    password = Sys.getenv("db_pwd"))
+  acpd_data <- dbReadTable(conn = conn,
+                       name = 'incidents')
+  
+  acpd_filtered <- acpd_data %>%
+    mutate(nearby = map2_dbl(.x = longitude,
+                             .y = latitude,
+                             .f = compute_distance_from_hq_in_km) <= 8.04672) %>%
```
Comment (Nosferican) [inline src/incidents_filtered.Rmd]: Make a commet # with the distance in miles.

File: src/incidents_filtered.Rmd [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    mutate(day = weekdays(as.Date(acpd_data$start)), 
+           hour = as.numeric(format(as.POSIXct(strptime(acpd_data$start, "%Y-%m-%d %H:%M:%S")), format = "%H"))) %>%
```
Comment (Nosferican) [inline src/incidents_filtered.Rmd]: Use `lubridate::as_date`

File: src/incidents_filtered.Rmd [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    mutate(crime_type = str_detect(string = description,
+                                   pattern ='(DRUNK|DUI|ASSAULT|POSSESSION|SEXUAL|DISORDERLY)')) %>%
+  
+      ## Filter for Time: party time = THURS-SUN 9PM-5AM (21:24, 1:5) but also include Cinco de Mayo and St. Patricks Day
+      filter(day %in% c("Thursday", "Friday", "Saturday", "Sunday") | grepl(paste(c("05-05 | 05-06 | 03-17"), collapse = "|"), start)) %>%
```
Comment (Nosferican) [inline src/incidents_filtered.Rmd]: Suggestion, instead of lame `grepl` use `str_detect`.

Comment (Nosferican) [reply src/incidents_filtered.Rmd]: And instead of `paste` use `str_c`.

File: src/incidents_filtered.Rmd [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+    mutate(day = weekdays(as.Date(acpd_data$start)), 
+           hour = as.numeric(format(as.POSIXct(strptime(acpd_data$start, "%Y-%m-%d %H:%M:%S")), format = "%H"))) %>%
+    mutate(crime_type = str_detect(string = description,
+                                   pattern ='(DRUNK|DUI|ASSAULT|POSSESSION|SEXUAL|DISORDERLY)')) %>%
+  
+      ## Filter for Time: party time = THURS-SUN 9PM-5AM (21:24, 1:5) but also include Cinco de Mayo and St. Patricks Day
+      filter(day %in% c("Thursday", "Friday", "Saturday", "Sunday") | grepl(paste(c("05-05 | 05-06 | 03-17"), collapse = "|"), start)) %>% 
+      filter(hour %in% c(1:5,21:24)) %>%
+      ## Filter for Location: filter to anything within a 5 mile radius ~ 8 km radius from metro origin
+      filter(nearby == TRUE) %>%
```
Comment (Nosferican) [inline src/incidents_filtered.Rmd]:
```suggestion
      filter(nearby) %>%
```

File: src/incidents_filtered.Rmd [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+      ## Filter for crime-type
+      filter(crime_type == TRUE)
+
+  acpd_filtered$nearby <- acpd_filtered$crime_type <- NULL
```
Comment (Nosferican) [inline src/incidents_filtered.Rmd]: ?

**Comment by Nosferican** [comment]
As we discussed, let's instead of have the indicator variables nightlife / nearby to filter within the shinny app instead of filtering the observations.

**Comment by Nosferican** [comment]
Instead of `dplyr` let use use `maditr` (i.e., `mutate` -> `dt_mutate`).

File: src/incidents_filtered.Rmd [hunk 9]
```diff
@@ -0,0 +1,72 @@
+---
+title: "incidents_filtered"
+author: "Kateryna Savchyn"
+date: "2019-06-11"
```
Comment (Nosferican) [inline src/incidents_filtered.Rmd]: 🥇

**Comment by Nosferican** [comment]
Is this ready to be reviewed?

**Comment by ksavchyn** [comment]
Yes!

**Pull request merged**